credentials/apps/credentials/utils.py

```
@@ -58,16 +61,95 @@ def get_programs():
 
     try:
         response = programs_api.programs.get()
+        results = response.get('results', no_programs)
+        cache.set(PROGRAMS_CACHE_KEY, results, settings.PROGRAMS_CACHE_TTL)
     except Exception:  # pylint: disable=broad-except
         log.exception('Failed to retrieve programs from the Programs API.')
         return no_programs
 
-    results = response.get('results', no_programs)
-    cache.set(PROGRAMS_CACHE_KEY, results, settings.PROGRAMS_CACHE_TTL)
-
     return results
 
 
+def get_organization(organization_key):
+    """ Get the organization from the edx-platform on behalf of the
+    credentials service user present in the edx-platform system.
+    Returned value is cached to avoid calling LMS each time a certificate is
+    viewed.
+
+    Arguments:
+        organization_key (str): Unique key of the organization for retrieval
+
+    Returns:
+        dict, representing organization data returned by the LMS.
+    """
+    no_organization = {}
+    # ensure that the 'organization_key' contains only ASCII characters
+    try:
+        organization_key.decode('ascii')
+    except UnicodeDecodeError:
+        log.exception('Invalid organization key %s.', organization_key)
+        return no_organization
+
+    cache_key = ORGANIZATIONS_CACHE_KEY + '.{}'.format(_make_hash(organization_key))
+    organization = cache.get(cache_key)
+
+    if organization is not None:
+        return organization
+
+    organizations_api = get_organizations_api_client()
+    if organizations_api is None:
+        return no_organization
```

How will the application respond to an empty dictionary instead of organization data? I would raise an exception and let the consuming code decide how to deal with this issue. The more you try to squash the exceptions, the more brittle the consuming code will become as it tries to work around the empty dictionary return value.